python/lsst/sims/photUtils/BandpassDict.py

```diff
@@ -0,0 +1,331 @@
+import copy
+import numpy
+from collections import OrderedDict
+from lsst.sims.photUtils import Bandpass, Sed
+
+__all__ = ["BandpassDict"]
+
+class BandpassDict(object):
+    """
+    This class will wrap an OrderedDict of Bandpass instantiations.
+
+    Upon instantiation, this class's constructor will resample
+    the input Bandpasses to be on the same wavelength grid (defined
+    by the first input Bandpass).  The constructor will then calculate
+    the 2-D phiArray for quick calculation of magnitudes in all
+    Bandpasses simultaneously (see the member method calcMagListFromSed).
+    """
+
+    def __init__(self, bandpassList, bandpassNameList):
+        """
+        @param [in] bandpassList is a list of Bandpass instantiations
+
+        @param [in] bandpassNameList is a list of tags to be associated
+        with those Bandpasses
+        """
+        self._bandpassDict = OrderedDict()
+        for bandpassName, bandpass in zip(bandpassNameList, bandpassList):
+
+            if bandpassName in self._bandpassDict:
+                raise RuntimeError("The bandpass %s occurs twice in your input " % bandpassName \
+                                   + "to BandpassDict")
+
+            self._bandpassDict[bandpassName] = copy.deepcopy(bandpass)
+
+        dummySed = Sed()
+        self._phiArray, self._wavelenStep = dummySed.setupPhiArray(self._bandpassDict.values())
+        self._wavelen_match = self._bandpassDict.values()[0].wavelen
```

Comment:
I think it would be really good to move away from the idea that we need to have the same set of wavelengths on every  band. This is because we will be dealing with objects whose seds are only known over finite wavelength ranges. Putting these objects at high redshifts means that there will be wavelengths in some bands for which the SED is not known, but can be reasonably assumed to be of similar or smaller magnitudes . Then it is possible to get fluxes for some bands and not others. Keeping a fixed wavelength set means declaring that we cannot do  the calculation for any band.
- Since this is the quantity that is used in matches, expanding this to an iterable means that the matches must also work on iterables.

maybe this should be a separate redesign task?